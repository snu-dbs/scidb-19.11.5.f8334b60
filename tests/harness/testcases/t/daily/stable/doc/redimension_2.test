--setup
--start-query-logging
create array raw <pos:int64, device:int64, val: float> [offset=1:*,5,0]
create array A<val:float>[device=1:5,5,0, pos=1:5,5,0]
create array B<val:double>[device=1:5,5,0, pos=1:5,5,0]

# Don't log, else dev box DOC_DATA expansion causes FILES_DIFFER.
--stop-query-logging
--aql load raw from '${DOC_DATA}/raw_2.txt' as 'csv:l'
scan(raw)  -- Inserted by scan_doctor
--start-query-logging

--test

# (The input file contains a data collision; "false" in these
# redimension() calls turns off strict collision checking.  Or, just
# get rid of the colliding input cell.)

redimension(raw, A, false)

# Wrong type because sum() aggregate always returns a double, regardless of whether input is float or double.
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_WRONG_ATTRIBUTE_TYPE "redimension(raw, A, false, sum(val) as val)"

# It works fine if the target attribute is a double.
redimension(raw, B, false, sum(val) as val)

--cleanup
remove(raw)
remove(A)
remove(B)
--stop-query-logging
