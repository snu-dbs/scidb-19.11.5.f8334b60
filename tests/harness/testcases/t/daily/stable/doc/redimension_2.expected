SCIDB QUERY : <create array raw <pos:int64, device:int64, val: float> [offset=1:*,5,0]>
Query was executed successfully

SCIDB QUERY : <create array A<val:float>[device=1:5,5,0, pos=1:5,5,0]>
Query was executed successfully

SCIDB QUERY : <create array B<val:double>[device=1:5,5,0, pos=1:5,5,0]>
Query was executed successfully

Query was executed successfully

{offset} pos,device,val
{1} 1,1,1.334
{2} 1,2,1.334
{3} 1,3,1.334
{4} 1,4,1.334
{5} 2,1,2.445
{6} 2,3,2.445
{7} 2,4,2.445
{8} 2,5,2.667
{9} 3,1,0.998
{10} 3,2,1.998
{11} 3,3,1.667
{12} 3,4,2.335
{13} 3,4,0.001
{14} 4,1,2.004
{15} 4,2,2.006
{16} 4,3,2.889
{17} 4,5,2.365
{18} 5,1,2.008
{19} 5,2,2.119
{20} 5,3,2.118
{21} 5,4,2.667
{22} 5,5,2.556

SCIDB QUERY : <redimension(raw, A, false)>
{device,pos} val
{1,1} 1.334
{1,2} 2.445
{1,3} 0.998
{1,4} 2.004
{1,5} 2.008
{2,1} 1.334
{2,3} 1.998
{2,4} 2.006
{2,5} 2.119
{3,1} 1.334
{3,2} 2.445
{3,3} 1.667
{3,4} 2.889
{3,5} 2.118
{4,1} 1.334
{4,2} 2.445
{4,3} 2.335
{4,5} 2.667
{5,2} 2.667
{5,4} 2.365
{5,5} 2.556

SCIDB QUERY : <redimension(raw, A, false, sum(val) as val)>
[An error expected at this place for the query "redimension(raw, A, false, sum(val) as val)". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_WRONG_ATTRIBUTE_TYPE. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_WRONG_ATTRIBUTE_TYPE.]

SCIDB QUERY : <redimension(raw, B, false, sum(val) as val)>
{device,pos} val
{1,1} 1.334
{1,2} 2.445
{1,3} 0.998
{1,4} 2.004
{1,5} 2.008
{2,1} 1.334
{2,3} 1.998
{2,4} 2.006
{2,5} 2.119
{3,1} 1.334
{3,2} 2.445
{3,3} 1.667
{3,4} 2.889
{3,5} 2.118
{4,1} 1.334
{4,2} 2.445
{4,3} 2.336
{4,5} 2.667
{5,2} 2.667
{5,4} 2.365
{5,5} 2.556

SCIDB QUERY : <remove(raw)>
Query was executed successfully

SCIDB QUERY : <remove(A)>
Query was executed successfully

SCIDB QUERY : <remove(B)>
Query was executed successfully

